### PR TITLE
Support --boms parameter for some commands

### LINKED ISSUE #91: BOM use-case for classpath and resolve commands
`classpath` and `resolve` commands could be extended to cover use-case with BOM.

Instead of GAV users would specify `artifact_GA` and `--bom=bom_GAV`

The downside is that the command description would become a bit more complicated

### LINKED ISSUE #95: Version is reported as 1.0
Mima version is always reported as 1.0

```
java -jar cli/target/cli-2.4.8-SNAPSHOT-uber.jar --version
1.0
```

```
mima --version
1.0
```

### PR DESCRIPTION
**Pull request by cstamas**
Commands supported:
- classpath
- graph
- resolve

Example invocation:
https://gist.github.com/cstamas/3ff4886f9f1d1a31a84f0fd105816754

Fixes #91 
Fixes #95 

---

Co-authored-by: Rostislav Svoboda <rsvoboda@redhat.com>

### CODE REVIEW

**Comment by rsvoboda** [comment]
Looking into it ;)

Here is first report for regression check
```
mima-snap classpath io.quarkus:quarkus-micrometer:3.7.1

java.lang.IllegalArgumentException: Bad artifact coordinates , expected format is <groupId>:<artifactId>[:<extension>[:<classifier>]]:<version>
    at org.eclipse.aether.artifact.DefaultArtifact.<init> (DefaultArtifact.java:78)
    at org.eclipse.aether.artifact.DefaultArtifact.<init> (DefaultArtifact.java:61)
    at eu.maveniverse.maven.mima.cli.ResolverCommandSupport.importBoms (ResolverCommandSupport.java:42)
    at eu.maveniverse.maven.mima.cli.Classpath.doCall (Classpath.java:43)
    at eu.maveniverse.maven.mima.cli.ResolverCommandSupport.call (ResolverCommandSupport.java:74)
    at eu.maveniverse.maven.mima.cli.ResolverCommandSupport.call (ResolverCommandSupport.java:19)
    at picocli.CommandLine.executeUserObject (CommandLine.java:2041)
    at picocli.CommandLine.access$1500 (CommandLine.java:148)
    at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent (CommandLine.java:2461)
    at picocli.CommandLine$RunLast.handle (CommandLine.java:2453)
    at picocli.CommandLine$RunLast.handle (CommandLine.java:2415)
    at picocli.CommandLine$AbstractParseResultHandler.execute (CommandLine.java:2273)
    at picocli.CommandLine$RunLast.execute (CommandLine.java:2417)
    at picocli.CommandLine.execute (CommandLine.java:2170)
    at eu.maveniverse.maven.mima.cli.Main.main (Main.java:41)
```

mima-snap created using `jbang app install --name mima-snap eu.maveniverse.maven.mima:cli:2.4.8-SNAPSHOT`

**Comment by rsvoboda** [comment]
There is no information about the resolved version of `org.hibernate.orm:hibernate-core`
I would like to know what version is dictated by the bom(s).

```
mima-snap resolve org.hibernate.orm:hibernate-core --boms=io.quarkus.platform:quarkus-bom:3.7.1

Resolved 44 artifacts
```

Comparison of released version and version based on apply-bom branch
```
mima resolve io.quarkus:quarkus-rest-client:3.7.1
Resolving io.quarkus:quarkus-rest-client:3.7.1

Resolved 221 artifacts
```

```
mima-snap resolve io.quarkus:quarkus-rest-client --boms=io.quarkus.platform:quarkus-bom:3.7.1

Resolved 221 artifacts
```

**Comment by rsvoboda** [comment]
mima should warn about unavailable BOM, the current implementation complains about bad coordinates of the artifact, should complain about bad coordinates of the BOM.

Example with fake io.quarkus.platform:quarkus-bom:3.7.195:
```
mima-snap resolve org.hibernate.orm:hibernate-core --boms=io.quarkus.platform:quarkus-bom:3.7.195 
BOM: 'io.quarkus.platform:quarkus-bom:3.7.195'
Error
java.lang.IllegalArgumentException: Bad artifact coordinates org.hibernate.orm:hibernate-core, expected format is <groupId>:<artifactId>[:<extension>[:<classifier>]]:<version>
    at org.eclipse.aether.artifact.DefaultArtifact.<init> (DefaultArtifact.java:78)
    at org.eclipse.aether.artifact.DefaultArtifact.<init> (DefaultArtifact.java:61)
    at eu.maveniverse.maven.mima.cli.ResolverCommandSupport.parseGav (ResolverCommandSupport.java:84)
    at eu.maveniverse.maven.mima.cli.Resolve.doCall (Resolve.java:64)
    at eu.maveniverse.maven.mima.cli.ResolverCommandSupport.call (ResolverCommandSupport.java:98)
    at eu.maveniverse.maven.mima.cli.ResolverCommandSupport.call (ResolverCommandSupport.java:20)
    at picocli.CommandLine.executeUserObject (CommandLine.java:2041)
    at picocli.CommandLine.access$1500 (CommandLine.java:148)
    at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent (CommandLine.java:2461)
    at picocli.CommandLine$RunLast.handle (CommandLine.java:2453)
    at picocli.CommandLine$RunLast.handle (CommandLine.java:2415)
    at picocli.CommandLine$AbstractParseResultHandler.execute (CommandLine.java:2273)
    at picocli.CommandLine$RunLast.execute (CommandLine.java:2417)
    at picocli.CommandLine.execute (CommandLine.java:2170)
    at eu.maveniverse.maven.mima.cli.Main.main (Main.java:41)
```

**Comment by cstamas** [comment]
Fixed all reported issues, with one remark: output of classpath is meant to be copy pasted. So use `-v` (verbose) and it will tell what version is being resolved.

File: cli/src/main/java/eu/maveniverse/maven/mima/cli/Resolve.java [hunk 1]
```diff
@@ -46,23 +44,33 @@ public final class Resolve extends ResolverCommandSupport {
...
+
         CollectRequest collectRequest = new CollectRequest();
-        collectRequest.setRoot(new Dependency(new DefaultArtifact(gav), JavaScopes.COMPILE));
+        collectRequest.setRoot(new Dependency(resolvedArtifact, JavaScopes.COMPILE));
         collectRequest.setRepositories(context.remoteRepositories());
+        collectRequest.setManagedDependencies(managedDependencies);
         DependencyRequest dependencyRequest =
                 new DependencyRequest(collectRequest, DependencyFilterUtils.classpathFilter(scope));
 
+        verbose("Resolving {}", collectRequest);
```
Comment (rsvoboda) [inline cli/src/main/java/eu/maveniverse/maven/mima/cli/Resolve.java:73]:
```suggestion
        info("Resolving {}", collectRequest.getRoot().getArtifact());
```

**Review by rsvoboda — CHANGES_REQUESTED** [review]
This will print the version without the need to enable verbose for `resolve`.

```
mima-snap resolve org.hibernate.orm:hibernate-core --boms=io.quarkus.platform:quarkus-bom:3.7.1

Resolving org.hibernate.orm:hibernate-core:jar:6.4.2.Final

Resolved 44 artifacts
```

**Review by rsvoboda — APPROVED** [review]

**Comment by cstamas** [comment]
@rsvoboda Thanks for contributing! 2.4.8 is out, JBang catalog updated as well.

**Pull request merged**